xblock/plugin.py

```diff
@@ -100,7 +100,7 @@ def select(identifier, all_entry_points):
             if select is None:
                 select = default_select
 
-            all_entry_points = list(pkg_resources.iter_entry_points(cls.entry_point, name=identifier))
+            all_entry_points = importlib.metadata.entry_points(group=cls.entry_point, name=identifier)
```

Comment:
```suggestion
            all_entry_points = list(importlib.metadata.entry_points(group=cls.entry_point, name=identifier))
```

According to the docs this returns [a collection](https://docs.python.org/3/library/importlib.metadata.html#entry-points), which means it is not necessarily a list, but it can be safely converted to list.  Since we use it as a list (`all_entry_points.append`) a few lines down, we should ensure that it is in fact a list.